### DISCUSSION TITLE
Given the current sheet, I want to get python code that take original sheet as input and produce output equivalent to current sheet.

### DISCUSSION THREAD

**Discussion by Awannaphasch2016** [saulpw/visidata#2460] [Q&A]
I know that visidata can show sql query (using show-sidebar command) that produce output equivalent to data represented in the current sheet. This works when open database in visidata 

Is there function that would show python code when open .csv (or other text file like) in visidata?

**Comment by midichef** [comment]
If I understand properly, you have text file data in Visidata, and want a way to "export" the data to a form you can process in a Python script running outside Visidata. Is that right?

First, I would use Visidata to convert the text file to JSON: `vd -b infile.csv outfile.json`
Then, if your Python script can use a file on the local filesystem:
```
#!/usr/bin/python
import json
with open('outfile.json') as fp:
    sheet_obj = json.load(fp)
```

If your script can't load local files, I would paste the JSON into the script:
```
#!/usr/bin/python3

# put the contents of the JSON file outfile.json between the triple quotes
json_string='''
[
{"col1": "one"},
{"col1": "two"},
{"col1": "three"}
]
'''

import json
sheet_obj = json.loads(json_string)
print(sheet_obj)
```

If this doesn't do what you want, let me know. It would be helpful to see a specific example of the exact input and final output Python code that you'd like to see.

**Comment by reagle** [comment]
@midichef, I don't think that's what they mean. I (might have) wondered the same thing: Can I reproduce what I've done in visidata outside of visidata, in something like pandas or polars? I think [Ibis](https://ibis-project.org/), a backend-neutral dataframe API, is supposed to allow that but apparently visidata hasn't been able to keep up with Ibis's breaking changes. See #2399 and #2410 .
